indico/modules/events/registration/client/js/form/fields/BooleanInput.jsx

```diff
@@ -35,36 +37,55 @@ function BooleanInputComponent({
   const formatPrice = useSelector(getPriceFormatter);
   const existingValue = useSelector(state => getFieldValue(state, fieldId));
 
-  const makeOnClick = newValue => () => {
-    if (value === newValue && !required) {
-      onChange(null);
-    } else {
-      onChange(newValue);
-    }
+  const handleSelect = targetValue => () => onChange(targetValue);
+  const handleUncheck = targetValue =>
+    required
+      ? () => {}
+      : () => {
+          if (targetValue === value) {
+            onChange(null);
+          }
+        };
+  const handleUncheckViaKeyboard = targetValue => {
+    const callback = handleUncheck(targetValue);
+    return evt => {
+      if (evt.code === 'Space' && value === targetValue) {
+        evt.preventDefault();
+        callback();
+      }
+    };
   };
 
   return (
     <div styleName="boolean-field">
-      <Button.Group id={id}>
-        <Button
-          id={`${id}-0`}
-          type="button"
-          active={!isPurged && value === true}
+      <label>
+        <input
+          type="radio"
+          name={`boolean-field-${id}`}
+          value="yes"
+          onChange={handleSelect(true)}
+          onClick={handleUncheck(true)}
+          onKeyDown={handleUncheckViaKeyboard(true)}
+          checked={!isPurged && value}
```

Comment:
```suggestion
          checked={!isPurged && value === true}
```

Otherwise the initial value may be `undefined`, resulting in an warning from React about a field changing from uncontrolled to controlled when choosing yes or no.